unit_test/test_spdm_callback/test_spdm_callback.c

```diff
@@ -0,0 +1,134 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#define LIBSPDM_MAX_CERT_CHAIN_SIZE 0x2000
+
+#include "spdm_unit_test.h"
+#include "library/spdm_common_lib.h"
+#include "spdm_crypt_ext_lib/spdm_crypt_ext_lib.h"
+
+/*read cert chain or root cert with dice extension*/
+bool libspdm_read_dice_certificate_chain(
+    void **data, size_t *size, bool is_cert_chain)
+{
+    bool res;
+    void *file_data;
+    size_t file_size;
+    char *file;
+
+    *data = NULL;
+    *size = 0;
+
+    if (is_cert_chain) {
+        file = "dice_cert/dice_cert_chain.bin";
+    } else {
+        file = "dice_cert/dice_root_cert.der";
+    }
+
+    res = libspdm_read_input_file(file, &file_data, &file_size);
+    if (!res) {
+        return res;
+    }
+
+    *data = file_data;
+    *size = file_size;
+
+    return true;
+}
+
+void libspdm_test_spdm_verify_cert_chain_callback_function(void **state)
+{
+    bool status;
+    libspdm_context_t *spdm_context;
+    uint8_t slot_id;
+    void *spdm_cert_chain_with_dicetcbinfo;
+    size_t spdm_cert_chain_size_with_dicetcbinfo;
+    void *spdm_root_cert_for_dicetcbinfo;
+    size_t spdm_root_cert_size_for_dicetcbinfo;
+
+    spdm_context = (void *)malloc(libspdm_get_context_size());
```

Comment:
The code should handle cases where malloc fails and returns NULL.

Reply:
Thanks. I have add the code to handle the failed case.